internal/worker/worker.go

```diff
@@ -126,17 +131,22 @@ func (w *worker) startJobLoop(ctx context.Context) <-chan error {
 	ech := make(chan error, w.limitation)
 
 	w.eg.Go(safety.RecoverFunc(func() (err error) {
+		defer close(ech)
 		eg, ctx := errgroup.New(ctx)
 		eg.Limitation(w.limitation)
+
 		limitation := make(chan struct{}, w.limitation)
 		defer close(limitation)
+
 		for {
 			select {
 			case <-ctx.Done():
+				var egErr error
 				if err = ctx.Err(); err != nil {
-					return errors.Wrap(eg.Wait(), err.Error())
+					egErr = eg.Wait()
+					return errors.Wrap(egErr, err.Error())
 				}
-				return eg.Wait()
+				return egErr
```

Comment:
I think that even when ctx.Err() is nil, you use eg.Wait()

Reply:
why you delete Wait?????

Reply:
please do not change goroutine shutdown logic with no proposals

Reply:
will fix it now!